### ISSUE TITLE
[API clean up] Verify test coverage of API endpoints already in use

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/openfoodnetwork#4072]
### What we should change and why (this is tech debt)
We need to make sure the API endpoints currently being used by external systems like zapier are covered with specs so that we don't delete or break them accidentally.

Ping @lin-d-hop can please you comment on this issue with the endpoints you think need to verified?

### Context
Part of API clean up. First step of making our API great.

**Comment by lin-d-hop** [comment]
Currently the only endpoint we are using is:
`/api/orders/${order_number}.json`

This returns:
https://gist.github.com/lin-d-hop/ad18e3568ad25069bd75ea15a32cd3ef

Obviously not all fields are used in the integration and this could be tidied quite a bit. However unless there is a good business case for tinkering this now perhaps it makes sense to leave as is until a more comprehensive API redesign?

**Comment by lin-d-hop** [comment]
The other endpoint that I am in conversations with users about (though not yet using) is:
`/api/orders.json`
Particularly with the parameters:
- completed_at_gt
- completed_at_lt
- distributor_id

**Comment by luisramos0** [comment]
Hi @lin-d-hop 
During my api clean up I mixed both entries above, I assumed /api/orders.json was needed and /api/orders/${order_number}.json could be needed soon.

In fact the /api/orders/${order_number}.json endpoint is coming from spree_api and disappears in https://github.com/openfoodfoundation/openfoodnetwork/pull/4057 :-(

I think we should move forward with this, with the current setup is quite easy to add a new endpoint to the API. and now we know where and how to do it :tada:

Can we open a new issue to re-add /api/orders/${order_number}.json after #4057 and go from there? For this issue we would need to know:
- What process is currently /api/orders/${order_number}.json?
- What fields are actually required from the huge payload referred above?
If the process is critical we should block #4057 from being merged until this new issue is also implemented.


btw, in terms of test coverage:
-  /api/orders/${order_number}.json this was being tested in spree but just basic validation that the endpoint is available and that it was implementing authorization (cannot see someone elses orders)
- /api/orders.json is tested in OFN

**Comment by lin-d-hop** [comment]
Apologies for brevity....

 /api/orders/${order_number}.json is critical for our Zapier integration to accounting systems - used by a number of UK users (not sure about other places).

Thus #4057 needs to be blocked until this is fixed.

Will add to the iissue now.

**Comment by RachL** [comment]
Outch. 

@luisramos0 

After the API clean up epic is done, would our API already be usable for external integration or is there more work to be done before that?

**Comment by luisramos0** [comment]
@RachL 👍 
discourse post for that: https://community.openfoodnetwork.org/t/is-the-ofn-api-ready-to-be-used-by-external-clients/1740

@lin-d-hop thanks, I'll start re-adding that endpoint right away 👍

**Comment by luisramos0** [comment]
@lin-d-hop if you put [this](https://gist.githubusercontent.com/lin-d-hop/ad18e3568ad25069bd75ea15a32cd3ef/raw/61f1cc63657b9a523467b3350d7ff345c240b7fc/order.json) into [this](http://jsonviewer.stack.hu/), it's easier to navigate ;-)

**Comment by luisramos0** [comment]
ok, the PR with the "new" endpoint is in code review now #4268

**Comment by luisramos0** [comment]
@lin-d-hop you can already verify in #4268 if the payload is good enough.

**Comment by luisramos0** [comment]
this will be closed by #4268

**Comment by luisramos0** [comment]
Closing this as 4268 is the only endpoint really used by zapier integrations and it was re-added with tests.
